metric_learn/base_metric.py

```diff
@@ -443,11 +444,22 @@ def score(self, quadruplets, y=None):
       points, or 2D array of indices of quadruplets if the metric learner
       uses a preprocessor.
 
-    y : Ignored, for scikit-learn compatibility.
+    y : array-like, shape=(n_constraints,) or `None`
+      Labels of constraints. y[i] should be 1 if
+      d(pairs[i, 0], X[i, 1]) is wanted to be larger than
+      d(X[i, 2], X[i, 3]), and -1 if it is wanted to be smaller. If None,
+      `y` will be set to `np.ones(quadruplets.shape[0])`, i.e. we want all
+      first two points to be closer than the last two points in each
+      quadruplet.
 
     Returns
     -------
     score : float
       The quadruplets score.
     """
-    return -np.mean(self.predict(quadruplets))
+    quadruplets = check_input(quadruplets, y, type_of_inputs='tuples',
```

Comment:
note that here quadruplets will be checked twice (once here, once in predict). This is because when I do `y = np.ones(quadruplets.shape[0])`, I want to be sure that I can do `quadruplets.shape[0]`, and otherwise an error message would be returned before (by the `check_input` method). I think it's fine isn't it ? I don't see any other solution to do so. Note that I also check at the same time `y` because I like the fact that `column_or_1d` check will be called on y (since it is not done in `accuracy_score`).